### ISSUE TITLE
[Issue]: Make TERMINATE higher priority over human_input_mode=ALWAYS 

### ISSUE THREAD

**Issue by hughlv** [microsoft/autogen#1086]
### Describe the issue

I'm defining my user_proxy as following to replicate the simple chat example from https://github.com/microsoft/autogen/blob/main/samples/simple_chat.py :

```py
user_proxy = UserProxyAgent(
    name="UserProxy",
    system_message="Hello AI",
    human_input_mode="ALWAYS",
    is_termination_msg=lambda x: x.get("content", "") and x.get("content", "").rstrip().endswith("TERMINATE"),
    max_consecutive_auto_reply=3,
    code_execution_config={
      "work_dir": "work_dir",
    },
)
```

the 'ALWAYS' mode works fine and when Assistant send back code snippet, simply pressing Enter will execute the code, also fine with `max_consecutive_auto_reply > 0`. The problem is when Assistant sends back message with `TERMINATE` at the end, it was ignored and still ask for human input. I have to input 'exit' to quit manually.

I think maybe it is better to make TERMINATE a higher priority over human_input_mode=ALWAYS.

### Screenshots and logs

<img width="710" alt="image" src="https://github.com/microsoft/autogen/assets/817110/92cff1e4-cfe3-45f9-849e-74d795d6a308">

**Comment by rickyloynd-microsoft** [comment]
@sonichi

**Comment by sonichi** [comment]
If you press enter in the end, it's supposed to also terminate. Is that not the case?

**Comment by hughlv** [comment]
> If you press enter in the end, it's supposed to also terminate. Is that not the case?

You're correct, pressing Enter will end the conversation. It's fine if it's by design. I can work around this.
